apps/framework-cli/src/mcp/server.rs

```diff
@@ -142,19 +134,40 @@ impl ServerHandler for MooseMcpHandler {
         _pagination: Option<PaginatedRequestParam>,
         _context: RequestContext<RoleServer>,
     ) -> Result<ListResourcesResult, ErrorData> {
-        Ok(embedded_docs::list_resources())
+        // Combine documentation resources and infrastructure resources
+        let mut doc_resources = embedded_docs::list_resources();
+        let infra_resources =
+            infra_resources::list_infra_resources(self.redis_client.clone()).await;
+
+        // Append infrastructure resources to documentation resources
+        doc_resources.resources.extend(infra_resources.resources);
+
+        Ok(doc_resources)
     }
 
     async fn read_resource(
         &self,
         request: ReadResourceRequestParam,
         _context: RequestContext<RoleServer>,
     ) -> Result<ReadResourceResult, ErrorData> {
-        embedded_docs::read_resource(&request.uri).ok_or_else(|| ErrorData {
-            code: ErrorCode::INVALID_PARAMS,
-            message: format!("Resource not found: {}", request.uri).into(),
-            data: None,
-        })
+        // Route based on URI scheme
+        if request.uri.starts_with("moose://infra/") {
+            // Handle infrastructure resources
+            infra_resources::read_infra_resource(&request.uri, self.redis_client.clone())
+                .await
+                .ok_or_else(|| ErrorData {
+                    code: ErrorCode::INVALID_PARAMS,
```

Comment:
am I missing something or this could error in a way that is not actually invalid params? Like say a transient network error?